sprocs/ip_to_mode.sql

```diff
@@ -33,5 +33,19 @@ BEGIN
     IF FOUND THEN
         mode := 'Exam';
     END IF;
+
+    -- Do we have an active PT reservation? If so, force mode to 'Exam'
+    PERFORM 1
+    FROM
+        pt_reservations AS r
+        JOIN pt_enrollments AS e ON (e.id = r.enrollment_id)
+    WHERE
+        e.user_id = ip_to_mode.authn_user_id
+        AND (date BETWEEN r.access_start AND r.access_end);
```

Comment:
Asking out of complete ignorance of how PT functions: do we need to check for deletion/"checked in" status, null dates, etc.? Basically, any of the things that we do for PS.

Reply:
Nope, it's strictly checking date between start/end times. Both of these times will be NULL if we are not checked in, so the BETWEEN evaluates to NULL which is not TRUE. When we check out, we set `access_end = now()` so any future checks will fail the BETWEEN check. We don't have a separate `checked_in` flag.

Note that BETWEEN cares about order, so "6 between 10 and 2" is FALSE. Doing "a BETWEEN b AND c" is just checking "b < a" and "a < c". See https://www.postgresql.org/docs/current/functions-comparison.html

Reply:
Gotcha! The way `BETWEEN` works with `NULL` seems very clever but does make intuitive sense. This all looks good to me!

Reply:
Yeah, it's maybe a bit too magic that BETWEEN handles all the NULL cases, but it does make the check very short and easy to write 😄 

Thanks for all the checking!